packages/tailwindcss/src/compat/apply-config-to-theme.ts

```diff
@@ -130,7 +130,13 @@ export function themeableValues(config: ResolvedConfig['theme']): [string[], unk
     }
 
     if (Array.isArray(value) && value.every((v) => typeof v === 'string')) {
-      toAdd.push([path, value.join(', ')])
+      if (path[0] === 'fontSize') {
```

Comment:
```suggestion
      if (path[0] === 'fontSize' && value.length === 2) {
```

I'd probably want to make sure this _only_ works for tuples.

Reply:
I think if we are to keep maximum compatibility with v3, we shouldn't restrict to 2 values. In v3, [it still emits `line-height`](https://play.tailwindcss.com/Twg4KzukfM) with > 2 values, though I'm not sure how likely this would have occurred in the real world.

Though thinking about this, perhaps we may need to check that there is **at least** two values for the `value[1]` reference?

Reply:
ah yep let's check for at least 2 entries then

Reply:
I've just tested again in v3 and an array value with one element, i.e. `2xl: ['2rem']` actually *unsets* the line-height. I'll assume we want to mirror this behavior as well in the compat layer.

Reply:
OK! Sure thing :)